### PR TITLE
Remove trailing space from emitLeadingComment

### PR DESCRIPTION
**Pull request by ubershmekel**
This will prevent emiting an extraneous trailing space on comments to
match https://eslint.org/docs/rules/no-trailing-spaces
This also removes a space that may have been originally there after a
comment but isn't necessary syntactically.

Fixes #
https://github.com/Microsoft/TypeScript/issues/18314

### CODE REVIEW

**Comment by ubershmekel** [comment]
@rbuckton I think your solution is the more correct but a bit more complex. Removing the extraneous space after single line comments seems like an easy win because it will always be followed by a newline.

Would you say this is good enough for now?

**Comment by ubershmekel** [comment]
To clarify - I've updated the implementation and test baseline since the review. This PR is pending a rereview.

File: src/compiler/comments.ts [hunk 1]
```diff
@@ -274,7 +274,7 @@ namespace ts {
             if (hasTrailingNewLine) {
                 writer.writeLine();
             }
-            else {
+            else if (_kind === SyntaxKind.MultiLineCommentTrivia) {
```
Comment (rbuckton) [inline src/compiler/comments.ts]: Remove the leading underscore (`_`) prefix from `_kind` here and in the parameter.

**Comment by rbuckton** [comment]
Thanks for the contribution!

**Pull request merged**